src/models/GameFilter.py

```diff
@@ -0,0 +1,10 @@
+from dataclasses import dataclass
+from typing import Optional
+
+
+@dataclass
+class GameFilter:
+    gameType: str = "boardgame"
+    minplayers: Optional[int] = None
```

Comment:
Do you think you should add the same `__post_init__` logic here that you have in `BoardGame` to coerce each item in the array to an integer if the data is provided?

Reply:
Hmmmmmm thinking through this more, maybe you are right.  Originally I did not mark this data class as frozen because I was thinking I care about modifying values for different parameters a user might pass in for filtering.  But now that you have me thinking, freezing the data class I think freezes the given **_instantiation_** of that data class!!  So once it's instantiated with the parameters from the user THAN it's frozen and can't be modified.  And that is aligned with the functionality I want for this class.

Reply:
I seem to miss small details like this often

Reply:
Although actually while I agree it should be marked as frozen, I don't think it needs the __post__init__. That was added because I was passing in values for variables that were strings when I wanted them to be ints and so used that to be able to turn them into ints.  But in this case I can be confident that these values are ints because that's coming from the user and not an API response.  And eventually when I add on the front end HTML a field for these parameters, I can add some handling there that only accepts int values and not strings.

Reply:
Gotcha, I haven't seen you test this but I would triple check on the backend that the types of the variables are in fact integers. HTTP is a strictly text-based protocol so it doesn't have the concept of sending _integers_ to an application, it just sends text. Some frameworks (like Django) allow you to write forms that handle the conversion of the input variables to a desired type I'm not sure if flask offers similar functionality.

Reply:
Ahhhhhh I see I see.  Yeah I'm totally making an assumption here because I haven't actually tested this.  Just assumed I would know the type being passed by a user and can be confident about it.  But if that's the case with HTTP than a user passing in "4" for max players would still be a string unless I changed it to an int somewhere.  

So it probably DOES make sense to add that post__init stuff here.  

I'm really glad we talked this out though because now it makes complete sense to me as why it should be.